streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -170,6 +170,8 @@ public boolean isValidTransition(final State newState) {
      */
     void closeDirty();
 
+    void updateInputPartitions(final Set<TopicPartition> topicPartitions);
```

Comment:
These two newly added functions are always called at the same time, I'd suggest we merge them into a single function, e.g. named `update(topicPartitions, topology)`.

Reply:
agree. changed to one function call